stonesoup/platform/base.py

```diff
@@ -1,37 +1,223 @@
     that moves according to a given :class:`~.TransitionModel`.
-    """
 
-    state = Property(State, doc="The platform state at any given point")
+    .. note:: Position and orientation are a read only properties in this class.
+    """
     transition_model = Property(
         TransitionModel, doc="Transition model")
 
-    def move(self, timestamp=None, **kwargs):
+    @property
+    def velocity(self) -> StateVector:
+        """Return the velocity of the platform.
+
+        Extracted from the state vector of the platform using the platform's
+        :attr:`velocity_mapping`. If the state vector is too short and does not contain the
+        elements specified in the :attr:`velocity_mapping` this raises an :class:`AttributeError`
+        """
+        try:
+            return self.state_vector[self.velocity_mapping]
+        except IndexError:
+            raise AttributeError('Velocity is not defined for this platform')
+
+    @property
+    def orientation(self) -> StateVector:
+        """Return the orientation of the platform.
+
+        This is defined as a 3x1 StateVector of angles (rad), specifying the sensor orientation in
+        terms of the counter-clockwise rotation around each Cartesian axis in the order
+        :math:`x,y,z`. The rotation angles are positive if the rotation is in the counter-clockwise
+        direction when viewed by an observer looking along the respective rotation axis,
+        towards the origin.
+
+        The orientation of this platform is defined as along the direction of its velocity, with
+        roll always set to zero (as this is the angle the platform is rotated about the velocity
+        axis, which is not defined in this approximation).
+
+        Notes
+        -----
+        A non-moving platform (``self.is_moving == False``) does not have a defined orientation in
+        this approximations and so raises an :class:`AttributeError`
+        """
+        if not self.is_moving:
+            raise AttributeError('Orientation of a zero-velocity moving platform is not defined')
+        velocity = self.velocity
+
+        if self.ndim == 3:
+            _, bearing, elevation = cart2sphere(*velocity.flat)
+            return StateVector([0, bearing, elevation])
+        elif self.ndim == 2:
+            _, bearing = cart2pol(*velocity.flat)
+            return StateVector([0, bearing])
+
+    @property
+    def is_moving(self) -> bool:
+        """Return the ``True`` if the platform is moving, ``False`` otherwise.
+
+        Equivalent (for this class) to ``all(v == 0 for v in self.velocity)``
+        """
+        # Note: a platform without a transition model can be given a velocity as part of it's
+        # StateVector. It just won't move
+        # This inconsistency is handled in the move logic
+        return np.any(self.velocity != 0)
+
+    def _set_position(self, value):
+        raise AttributeError('Cannot set the position of a moving platform')
```

Comment:
Is there a reason to limit this?

Reply:
Looking at the waypoint navigation work it might be sensible to allow it to be set externally, for example if you want to take real world movement data and run it through a simulator

Reply:
The reason I limited it was the assumption that platforms with transition models should only move via the transition model. This may be faulty, and happy to change if sensible. 

You can always set the state_vector directly if required, but again very willing to change this if wanted.

Reply:
I think the compromise solution is to check for `transition_model is None` at this point, and raise an error only if the transition model is set. This is what I have implemented in a new version to be pushed shortly.